src/extensions/default/PhpTooling/CodeHintsProvider.js

```diff
@@ -115,7 +119,12 @@ define(function (require, exports, module) {
                     }
                 }
 
-                var filteredHints = filterWithQueryAndMatcher(res, self.query);
+                var filteredHints = [];
+                if (hasIgnoreCharacters || (isExplicitInvokation && !trimmedQuery)) {
+                    filteredHints = filterWithQueryAndMatcher(res, "");
+                } else {
+                    filteredHints = filterWithQueryAndMatcher(res, self.query);
```

Comment:
do we need to filter hints. isn't server itself sending filtered hints based on the query??

Reply:
Filtering here isn't required for creating stringRanges, and matchGoodness, which are then used to highlight the matched text and sort the things accordingly as per the query.

Reply:
do we have any known scenario where hints returned from server is not a good match with query?

Reply:
if the purpose is to create  stringRanges, and matchGoodness, then why this function removes some of hints from hintlist retured from server??

Reply:
Can we make change here that none of the hints returned from server should be removed?
If we can do this then we will not have to worry about ignoreCharacters list

Reply:
I am not sure I understand. The characters in ignoreList are from the query, not the hints. In essence, I am actually doing the same thing you have suggested. The query is being saved on the Brackets side for a session. So when a user types "se" the results are then filtered by calculating the distance of the hint from the subsequence "se". The results are then sorted by the distance. When a user types something like "->" which is not part of the hint but still invokes hints, I am just not using "->" to filter out the hints, rather preserving the order of the hints as they come.

Reply:
@shubhsnov 
I feel same thing  can be done without maintaining an ignore characters list.

make change in filterWithQueryAndMatcher function.
Pass a boolean parameter(dontRemoveIfNoMatch) to this function. if this third parameter is true none of the hints should be removed even if hints has no match with query.
call this "filterWithQueryAndMatcher" for server hints with dontRemoveIfNoMatch as true.
and call this function separately for static hints with dontRemoveIfNoMatch  as false which is default.

lets think if we can do this?

Reply:
@niteskum `filterWithQueryAndMatcher` implementation shouldn't be changed as there are multiple extensions using this fn. This is more generic implementation which shouldn't be altered for code hint listing use cases.